lib/puppet/util/windows/process.rb

```diff
@@ -204,6 +204,37 @@ def elevated_security?
   end
   module_function :elevated_security?
 
+  def windows_major_version
+    ver = nil
+
+    FFI::MemoryPointer.new(OSVERSIONINFO.size) do |os_version_ptr|
+      os_version = OSVERSIONINFO.new(os_version_ptr)
+      os_version[:dwOSVersionInfoSize] = OSVERSIONINFO.size
+
+      result = GetVersionExW(os_version_ptr)
+
+      if result == FFI::WIN32_FALSE
+        raise Puppet::Util::Windows::Error.new(
+          "GetVersionEx(#{os_version_ptr})")
```

Comment:
Updated to remove ptr from message.  Note we still have at least these guys kicking around that use ptrs in error messages.

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/file.rb#L172-L173

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L51-L52

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L89-L90

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L108-L109

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/process.rb#L119-L120